src/lib/components/tabsv2/Tabsv2.component.js

```diff
@@ -0,0 +1,402 @@
+      if (selectedTabMeta.left < tabsMeta.left) {
+        // left side of button is out of view
+        const nextScrollStart =
+          tabsMeta.scrollLeft + (selectedTabMeta.left - tabsMeta.left);
+        scroll(nextScrollStart, { animation });
+      } else if (selectedTabMeta.right > tabsMeta.right) {
+        // right side of button is out of view
+        let nextScrollStart =
+          tabsMeta.scrollLeft + (selectedTabMeta.right - tabsMeta.right);
+        if (
+          scrollButtonEndRef.current &&
+          ((displayScroll.start && !displayScroll.end) ||
+            (!displayScroll.start && displayScroll.end))
+        ) {
+          nextScrollStart += scrollButtonEndRef.current.getBoundingClientRect()
+            .width;
+        }
+        scroll(nextScrollStart, { animation });
+      }
+    },
+    // we do not want to re-trigger this function if display.start / display.end are modified
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [getTabsMeta],
+  );
+
+  const updateScrollButtonState = useCallback(() => {
+    if (tabsRef.current) {
+      const { scrollLeft, scrollWidth, clientWidth } = tabsRef.current;
+      let showStartScroll = scrollLeft > 1;
+      let showEndScroll = scrollLeft < scrollWidth - clientWidth - 1;
+
+      if (
+        showStartScroll !== displayScroll.start ||
+        showEndScroll !== displayScroll.end
+      ) {
+        setDisplayScroll({ start: showStartScroll, end: showEndScroll });
+      }
+    }
+  }, [displayScroll.end, displayScroll.start]);
+
+  useEffect(() => {
+    window.addEventListener('resize', updateScrollButtonState);
+    return () => {
+      window.removeEventListener('resize', updateScrollButtonState);
+    };
+  }, [updateScrollButtonState]);
+
+  const handleTabsScroll = useMemo(() => () => updateScrollButtonState(), [
+    updateScrollButtonState,
+  ]);
+
+  useEffect(() => {
+    updateScrollButtonState();
+  });
+
+  // scroll to selected tab on first render
+  useEffect(() => {
+    scrollSelectedIntoView(true);
+  }, [scrollSelectedIntoView, selectedTabIndex]);
+
+  const handleKeyDown = (event) => {
+    const list = tabsListRef.current;
+    const ownerDocument = (list && list.ownerDocument) || document;
+    const currentFocus = ownerDocument.activeElement;
+    const role = currentFocus.getAttribute('role');
+    if (role !== 'tab') {
+      return;
+    }
+
+    switch (event.key) {
+      case 'ArrowLeft':
+        event.preventDefault();
+        moveFocus(list, currentFocus, previousItem);
+        break;
+      case 'ArrowRight':
+        event.preventDefault();
+        moveFocus(list, currentFocus, nextItem);
+        break;
+      case 'Home':
+        event.preventDefault();
+        moveFocus(list, null, nextItem);
+        break;
+      case 'End':
+        event.preventDefault();
+        moveFocus(list, null, previousItem);
+        break;
+      default:
+        break;
+    }
+  };
```

Comment:
In order to simplify the component maintainability I would move this logic to a custom hook : 
`const {scrollButtonEndRef, scrollButtonStartRef, tabsListRef, tabsRef} = useScrollingTabs()`

Reply:
Also as this code is be copied from https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/Tabs/Tabs.js I think it would be good to add a comment explaining this and copying the license of material-ui before the implementation of the hook

Reply:
To simplify the custom hook can be mutualised with the animation functions in the same file.